### ISSUE TITLE
Realtime Database After Event Missing values 

### ISSUE THREAD

**Issue by JJ11teen** [firebase/firebase-functions-python#122]
Hello! 👋

There is a bug in the logic to generate the "after" state of a (realtime) database change. The bug appears to be in the below lines of code, only a shallow copy/merge is performed but a deep one is needed.

https://github.com/firebase/firebase-functions-python/blob/14676e5156e790c2f25a5e6df7f6a4ff24888188/src/firebase_functions/db_fn.py#L92-L94

As an example of a failure case, I had just changed `allocation/change` from `F` to `G` in the following:
![image](https://github.com/firebase/firebase-functions-python/assets/1443174/621e7907-2155-4837-a308-59593c081a66)

However I have a listener setup on the root object of the screenshot and the `event.data` object that is generated by that change was:
```
Change(
    before={'allocation': {'change': 'F', 'role': 'admin', 'third': 3}, 'details': <redacted>},
    after={'allocation': {'change': 'G'}, 'details': <redacted>})
```

It should be:
```
Change(
    before={'allocation': {'change': 'F', 'role': 'admin', 'third': 3}, 'details': <redacted>},
    after={'allocation': {'change': 'G', 'role': 'admin', 'third': 3}, 'details': <redacted>})
```

Let me know if you need any more info.

Thanks!

**Comment by exaby73** [comment]
Hello @JJ11teen. Unfortunately, I am having some trouble reproducing this issue. Are you able to reproduce this on the Firebase emulator? Could you provide some reproduction code that I may test against?

**Comment by JJ11teen** [comment]
Hey @exaby73, unfortunately I don't have time to create a complete minimal reproducible code for you. Is there any other way I can help you understand the problem I am describing? If you look at the code I have pointed out in `db_fn.py` it should make sense why the issue is arising without needing to reproduce the problem on your side - The code `{**before, **after}` is a shallow merge of the before & after dictionaries ("after" at this point is actually the delta). A deep merge is required to accurately construct an "after" state.


For anyone else following along I've worked around this issue in the mean time by (re)querying the current state of the DB to determine the after state.
